fx/graph_profiler.py

```diff
@@ -0,0 +1,525 @@
+import logging
+import statistics
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import tabulate
+import torch
+from graph_profiler_utils import (MEM_LIMIT, DEVICE, BiDict, GraphProfiler, GraphType,
+                                IntNodeInfo, NodeInfo, ProfileMode,
+                                TensorStatus, get_tensor_stat,
+                                profile_mode_dict)
+from torch.fx import GraphModule, Interpreter, Node
+from torch.fx.node import map_arg
+
+
+class GraphProfiler(Interpreter):
```

Comment:
Hi @sanketpurandare , thanks for your contribution! Things like GraphProfiler and ProfileEngine look complicated and generic. Can we move them to pytorch repo? 
In general, we should put some simple scripts with intuitive READMEs in pytorch/examples for new users to get started. e.g. a script demonstrating how to use GraphProfiler and a README showing the CLI to run the example